### PR TITLE
add work object to data layer

### PR DESCRIPTION
**Pull request by gestchild**
References #4341 

Adds the work object to the data layer, so we can make use of it in tag manager

### CODE REVIEW

File: catalogue/webapp/pages/work.js [hunk 1]
```diff
@@ -50,6 +50,7 @@ export const WorkPage = ({ work }: Props) => {
   };
 
   useEffect(() => {
+    window.dataLayer && window.dataLayer.push({ work: work });
```
Comment (davidpmccormick) [inline catalogue/webapp/pages/work.js]: When do we use `if` statements nowadays? I think I'd probably have written one here, but not sure why.

**Comment by jamesgorrie** [comment]
Where is the `dataLayer` set?
I smell a bit of GA magic.

**Comment by jamesgorrie** [comment]
Given that the dataLayer is an array - how would @hthair know which index is the thing she is looking for?

**Comment by davidpmccormick** [comment]
it's an object innit?

**Comment by jamesgorrie** [comment]
> it's an object innit?

If so - we really shouldn't be running `.push` on it.

**Comment by gestchild** [comment]
It's an array :)

**Comment by gestchild** [comment]
> Given that the dataLayer is an array - how would @hthair know which index is the thing she is looking for?

In Tag Manager you set the Data Layer Variable Name field to , in this case, 'work'

**Comment by davidpmccormick** [comment]
First line of the TagManager data layer docs is very confusing in that case.

'A data layer is a JavaScript object that is used to pass information from your website to your Tag Manager container'

**Comment by gestchild** [comment]
> Where is the `dataLayer` set?
> I smell a bit of GA magic.

Yeah, the Tag Manager script does the magic

**Comment by jamesgorrie** [comment]
It still feels a little like we might abstract this out and have a nice way of setting the `dataLayer` and have the side-effect controlled elsewhere so when we need this later it's a matter of setting a prop on the PageLayout or something, but I think I am being precious.

**Pull request merged**